### PR TITLE
Revise the options about kinds, fields, and extras

### PR DESCRIPTION
**Pull request by masatake**

### CODE REVIEW

**Comment by masatake** [comment]
"letters" and "names" are used so freqently in our man pages. So I would like to introduce more distinguishable words for them.

For letter, I introduced "one-letter" or "one-letter flag".
About "names", I would like to introduce "long-name" or "long-name flag".

**Comment by masatake** [comment]
The sed command line synthesized  by make is the same as that in GNU/Linux.
So I guess t his is not about the implementation of make.

This may be about the implementation of sed.
https://riptutorial.com/ja/sed/topic/9436/bsd---macos-sed%E5%AF%BEgnu-sed%E5%AF%BEposix-sed%E4%BB%95%E6%A7%98 (in Japanese)

**Comment by masatake** [comment]
When building ctags executable, we support using bsd make.
However, generating documents is done in before buidling.
So we can assume GNU make is there.

**Comment by k-takata** [comment]
Oh, BSD sed doesn't support `\|`. It is critical...

**Comment by k-takata** [comment]
Is it better to stop executing `make update-docs` on mac?

**Comment by k-takata** [comment]
If we revert #2347, we can use BSD sed, but you may not like it. ;-)

I confirmed that the same results with macOS are produced when adding `--posix` option to GNU sed.

**Comment by masatake** [comment]
> Is it better to stop executing make update-docs on mac?

My understanding is that person using Mac cannot develop u-ctags.
If my understanding is correct, I would like to make "make update-docs" workable even on Mac.

Currently, you make single complicated pattern in reppat.
I wonder we can convert it to multiple patterns like:
```
  sed -e pat1 -e pat2 -e pat3 ....
```
So we can avoid using '\|'.

**Comment by masatake** [comment]
As you guessed, I like #2347. 
This is good time for me to study what you did.
I will try to fix this issue by myself.

**Comment by k-takata** [comment]
BSD sed doesn't support `\<`. This is also a problem.

**Comment by masatake** [comment]
```
commit 4fe409f9aba2e631e048e0b874b03eb34b4df1c7 (HEAD -> misc-doc-fix)
Author: Masatake YAMATO <yamato@redhat.com>
Date:   Mon Dec 2 17:22:08 2019 +0900

    docs(web), docs(man): don't use '\|' in a pattern for sed when making hyperlinks
    
    sed implementation in MacOS doesn't understand '\|'.
    To avoid using '\|', this change expands 'A\|B' to "-e A -e B".
    
    Signed-off-by: Masatake YAMATO <yamato@redhat.com>

diff --git a/man/Makefile b/man/Makefile
index 74ec29de..d191c626 100644
--- a/man/Makefile
+++ b/man/Makefile
@@ -64,14 +64,26 @@ man: $(man_pages)
 html: $(html_pages)
 pdf: $(pdf_pages)
 
-empty :=
-space := $(empty) $(empty)
-reppat := \<\(${subst ${space},\|,${subst .,(,${addsuffix ),${man_pages}}}}\)
+#
+# Convert the names of man page files to man page citations.
+#
+# ctags.1.rst tags.5.rst ...
+# => ctags(1) tags(5) ...
+#
+cites := ${subst .,(,${addsuffix ),${man_pages}}}
+
+#
+# Convert the man page citations to sed patterns for making hyperlinks.
+#
+# ctags(1) tags(5) ...
+# => -e 's/\<ctags(1)/:ref:`& <&>`/g' -e 's/\<ctags(5)/:ref:`& <&>`/g' ...
+#
+reppat := $(foreach m,$(cites),-e 's/\<$(m)/:ref:`& <&>`/g')
 
 update-docs: $(doc_files)
 
 ../docs/man/%.rst: %.rst
-   sed -e 's/$(reppat)/:ref:`& <&>`/g' < $< > $@
+   sed $(reppat) < $< > $@
 
 %.rst: %.rst.in
 ifeq ($(QUICK),)
```

**Comment by masatake** [comment]
> BSD sed doesn't support \<. This is also a problem.

This is a critical issue. Is there anyway to avoid using it?

**Comment by k-takata** [comment]
I used `\<` to avoid replacing `.. _ctags(1):` in top of the rst files. If it can be skipped, `\<` is not needed.

File: man/Makefile [hunk 1]
```diff
@@ -64,14 +64,26 @@ man: $(man_pages)
 html: $(html_pages)
 pdf: $(pdf_pages)
 
-empty :=
-space := $(empty) $(empty)
-reppat := \<\(${subst ${space},\|,${subst .,(,${addsuffix ),${man_pages}}}}\)
+#
+# Convert the names of man page files to man page citations.
+#
+# ctags.1.rst tags.5.rst ...
```
Comment (k-takata) [inline man/Makefile]: This line should be:
`# ctags.1 tags.5 ...`

**Comment by masatake** [comment]
Though I did nothing for `\<`, the tests are passed.

**Comment by k-takata** [comment]
Oh, does recent version of mac's sed support `\<`?

**Comment by masatake** [comment]
> Oh, does recent version of mac's sed support \<?

Maybe yes.
About Mac, what we can do is only believing the result of travis.

File: man/Makefile (hunk 1 above)
Comment (masatake) [reply man/Makefile]: Thank you. Updated.

**Comment by masatake** [comment]
4a962cf is needed other pull requests.
I will commit it separately.

**Comment by masatake** [comment]
TODO

* the other man pages
* long-name

**Comment by masatake** [comment]
Introducing the terms, "one-letter" and "long-name", to the other man pages is not done yet.
But I will merge this pull request. I will update the man pages later.

**Pull request merged**